src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/client/HomeOfficeCaseStatusClientTest.java

```diff
@@ -0,0 +1,160 @@
+
+@ExtendWith(MockitoExtension.class)
+public class HomeOfficeCaseStatusClientTest {
+
+    private final String someEndpoint = "some-endpoint";
+    private final String someUri = "some-uri";
+    private final String someResponse = "some-response";
+    private final String someHomeOfficeReference = "some-ho-reference";
+    @Mock
+    private RestTemplate restTemplate;
+    @Mock
+    private HttpClientErrorException httpClientErrorException;
+    @Mock
+    private AsylumCase asylumCase;
+    @Mock
+    private HomeOfficeApiUtil homeOfficeApiUtil;
+    @Mock
+    private ObjectMapper objectMapper;
+    @Captor
+    private ArgumentCaptor<HttpEntity<Map<String, List>>> httpEntityArgumentCaptor;
+    private HomeOfficeCaseStatusClient homeOfficeCaseStatusClient;
+
+    @BeforeEach
+    public void setUp() throws IOException {
+
+        homeOfficeCaseStatusClient = new HomeOfficeCaseStatusClient(
+            someEndpoint,
+            someUri,
+            restTemplate,
+            homeOfficeApiUtil);
+    }
+
+    @Test
+    public void returns_case_data_from_home_office_api_call_for_sample_response() {
+
+        when(asylumCase.read(AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER))
+            .thenReturn(Optional.of(someHomeOfficeReference));
+
+        doReturn(someResponse).when(restTemplate).postForObject(
+            anyString(),
+            any(HttpEntity.class),
+            eq(String.class));
+
+        String actualResponse = homeOfficeCaseStatusClient.getCaseStatus(asylumCase);
+
+        verify(restTemplate, times(1))
+            .postForObject(
+                eq(someEndpoint + someUri),
+                httpEntityArgumentCaptor.capture(),
+                eq(String.class));
+
+        assertThat(actualResponse).isEqualTo(someResponse);
+    }
+
+    @Test
+    public void handles_http_exception() {
+
+        when(asylumCase.read(AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER))
+            .thenReturn(Optional.of(someHomeOfficeReference));
+
+        when(restTemplate.postForObject(
+            anyString(),
+            any(HttpEntity.class),
+            any())).thenThrow(httpClientErrorException);
+
+        when(httpClientErrorException.getResponseBodyAsString())
+            .thenReturn("some-response-body");
+
+        assertThatThrownBy(() -> homeOfficeCaseStatusClient.getCaseStatus(asylumCase))
+            .isExactlyInstanceOf(HomeOfficeResponseException.class)
+            .hasMessage("some-response-body")
+            .hasCause(httpClientErrorException);
+    }
+
+    @Test
+    public void create_requestStr_should_return_valid_json_request_string_with_header() {
+        when(asylumCase.read(AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER))
+            .thenReturn(Optional.of(someHomeOfficeReference));
+
+        HttpEntity request = homeOfficeCaseStatusClient.createRequestString(asylumCase);
+
+        assertNotNull(request);
+        assertThat(request.getBody().toString()).isEqualTo("{"
```

Comment:
This will need changing when you model the request/response properly

Reply:
Will remove once model is done